### DISCUSSION TITLE
Language in html_document - toc levels in learnr::tutorial - themes

### DISCUSSION THREAD

**Discussion by BB-Diesunddas** [rstudio/learnr#813] [Q&A]
Hi,

I am experienced in R, but new to learnr. I really like the concept. Now I am experimenting to prepare teaching material (for several university courses with 100+ participants).

Here are some questions, to which I didn't find a solution:
- if I use the output "html_document" instead of "learnr::tutorial", how do I change the language of the interactive code boxes? For the latter it works perfectly via yaml "language: xyz", but how to do it for "html_document"?
- one reason for using "html_document" is that its (floating) toc allows multiple levels of headings, so one can structure & navigate a large document much better. Is there a way to allow mutliple levels in "learnr::tutorial", it seems currently only to work with ##?
- There are several themes offered, is there a site or "themebrowser" to compare these side by side? (instead of everyone just diy)

Thank you for your work and help!

**Comment by gadenbuie** [comment]
> * if I use the output "html_document" instead of "learnr::tutorial", how do I change the language of the interactive code boxes? For the latter it works perfectly via yaml "language: xyz", but how to do it for "html_document"?

In this case you should be able to use the `tutorial.language` option, e.g. `options(tutorial.language = "fr")`. Alternatively, `knitr::opts_knit$set(tutorial.language = "fr")` should also work.


> * Is there a way to allow mutliple levels in "learnr::tutorial", it seems currently only to work with ##?

No there isn't, unfortunately.



> * There are several themes offered, is there a site or "themebrowser" to compare these side by side? (instead of everyone just diy)

This is a great idea but it doesn't currently exist. For now I'd recommend using a small, simple tutorial to step through themes until you find one you like.

**Reply by BB-Diesunddas** [reply]
Thank you for your reply. Still the language switch seems not to work for me in `html_document`. Here is a minimal example (trying all possible positions for the option without success):

````
---
title: "Language for html_document"
output:
#  learnr::tutorial:
#    language: fr
  html_document   
runtime: shiny_prerendered 
---


```{r setup, include=FALSE}
options(tutorial.language = "fr")
knitr::opts_knit$set(tutorial.language = "fr")
library(learnr)
options(tutorial.language = "fr")
knitr::opts_knit$set(tutorial.language = "fr")
```


```{r test, exercise = TRUE}

```
````

The generated document shows still "Start Over" and "Run Code" at the coding-box.
